packages/integration-aws-node/README.md

```diff
@@ -0,0 +1,105 @@
+# @backstage/integration-aws-node
+
+This package providers helpers for fetching AWS account credentials
+to be used by AWS SDK clients in backend packages and plugins.
+
+## Integrate new plugins
+
+Backend plugins can provide an AWS ARN or account ID to this library in order to
+retrieve a credentials provider for the relevant account that can be fed directly
+to an AWS SDK client.
+AWS SDK for Javascript V3 must be used.
+
+```typescript
+const awsCredentialsProvider = DefaultAwsCredentialsProvider.fromConfig(config);
+
+// provide the account ID explicitly
+const creds = await awsCredentialsProvider.getCredentials({ accountId });
+// OR extract the account ID from the ARN
+const creds = await awsCredentialsProvider.getCredentials({ arn });
```

Comment:
Does this support some form of incomplete ARN, for example, I want to have access to a specific S3 bucket, or S3 buckets in general, but I don't know which account they're in? Or can the account always be derived from the resource you're trying to access? (haven't used AWS too much 😅)

Reply:
S3 bucket ARNs are the only AWS ARNs that don't have an account ID in them, so every other resource type can derive the account ID from the ARN.

For the S3 bucket case, I would expect something like the S3 integration to have some static config with the account ID and then call `awsCredentialsProvider.getCredentials({ accountId });`
```yaml
integrations:
  awsS3:
    - host: amazonaws.com

      # Old way of configuring the integration
      # accessKeyId: ${AWS_ACCESS_KEY_ID}
      # secretAccessKey: ${AWS_SECRET_ACCESS_KEY}

      # Proposed new way of configuring the integration.  This can also be omitted to fall back to the default creds.
      accountId: '123456789012'

aws:
  accounts:
    - accountId: '123456789012'
      roleName: 'my-iam-role-name'
```